src/bin/render.rs

```diff
@@ -0,0 +1,23 @@
+//! Test tool to render a LinkerScript to `stdout`
+//!
+//! This might not represent a linker script that can be used on a
+//! device! But, it may help with visually inspecting the output.
+
+use imxrt_rt_gen::*;
+use std::io;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let mut ls = LinkerScript::<u32>::new();
+    let flash = ls.region(FLASH, 0x0, 512).unwrap();
+    let ram = ls.region(RAM, 0x20000000, 128).unwrap();
+    ls.stack(ram.clone()).unwrap();
+    ls.heap(ram.clone()).unwrap();
+    ls.boot_config(512, "fcb", flash.clone()).unwrap();
+    ls.vector_table(ram.clone(), Some(flash.clone())).unwrap();
+    ls.text(flash.clone(), None).unwrap();
+    ls.data(false, ram.clone(), Some(flash.clone())).unwrap();
+    ls.rodata(false, flash.clone(), None).unwrap();
+    ls.bss(false, ram.clone(), Some(flash.clone())).unwrap();
```

Comment:
Any reason to favor the `.clone()` calls over immutable borrows?

```rust
ls.bss(false, &ram, Some(&flash)).unwrap();
```

is fewer characters to type.

Reply:
Immutable borrows are fine, I had a copyable integer id being passed around before switching to a string identifier, one of those times where I wish I had a copyable string type, these strings aren't long.